Makefile

```diff
@@ -82,14 +82,18 @@ build-e2e:
 	$(DOCKER_CMD) go test -c -o bin/machines.test github.com/openshift/cluster-api-provider-libvirt/test/machines
 
 .PHONY: test-e2e
-e2e: images build-e2e e2e-provision ## Run end-to-end test
-	hack/test-e2e.sh
-	hack/packet-provision.sh destroy
+test-e2e: images build-e2e e2e-provision ## Run end-to-end test
+	hack/test-e2e.sh || $(MAKE) e2e-clean
+	$(MAKE) e2e-clean
```

Comment:
If the e2e fails, would be great to keep the environment for inspection. So more like have the `make e2e-clean` called explicitly in the CI/locally by a developer.